src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AlleleFilteringUnitTest.java

```diff
@@ -298,5 +298,70 @@ public void testNotFilterLoneWeakAllele(){
 
     }
 
+    @Test //check that we filter strong allele with high SOR 
```

Comment:
I think this test might test what you want here. I would recommend adding a unit test to the AlleleFiltering.identifyBadAlleles() (made package protected and marked with @visiblefortesting) and constructing a specific case that triggers this bug and explaining that its capturing the state where the SOR and Qual based filtering sorting are out of step and you are still fitlering appropriately.

Reply:
@jamesemery - good idea, added test!